OCPBUGS-37089#adding Important text (#89952)
openshift/openshift-docs: modules/installation-cloudformation-worker.adoc

// * installing/installing_aws/installing-aws-user-infra.adoc
// * installing/installing_aws/installing-restricted-networks-aws.adoc

:_mod-docs-content-type: REFERENCE
[id="installation-cloudformation-worker_{context}"]
= CloudFormation template for compute machines

> You can use the following CloudFormation template to deploy the compute machines that you need for your {product-title} cluster.

Reviewer: ```suggestion
You can deploy the compute machines that you need for your {product-title} cluster by using the following CloudFormation template.
```

Reviewer: https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#writing-steps:~:text=If%20you%20orient%20the%20user%20to%20the%20location

.CloudFormation template for compute machines
[%collapsible]
====
[source,yaml]
----
include::https://raw.githubusercontent.com/openshift/installer/release-4.19/upi/aws/cloudformation/06_cluster_worker_node.yaml[]
----
====
[IMPORTANT]
====

> When you use the CloudFormation template for the control plane machines, it provisions all three control plane machines with a single stack. However, when you use the CloudFormation template to deploy the compute machines, you must create the number of stacks based on the number that you defined in the `install-config.yaml` file. Each stack is provisioned one time only for each machine. To provision a new compute machine, you must change the stack name.

Reviewer: ```suggestion
When you use the CloudFormation template for the control plane machines, the template provisions all the three control plane machines with a single stack; however, when you use the CloudFormation template to deploy the compute machines, you must create the number of stacks based on the number that you defined in the `install-config.yaml` file. Each stack is provisioned once for each machine. To provision a new compute machine, you must change the stack name.
```

Reviewer: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#:~:text=one%20word%2C%20unhyphenated.-,once%20adverb,-Use%20only%20to

Reviewer: https://www.ibm.com/docs/en/ibm-style?topic=punctuation-semicolons#semicolons-between-independent-clauses

Reviewer: https://www.ibm.com/docs/en/ibm-style?topic=grammar-pronouns#ambiguous-pronoun-references

====

> . To create the CloudFormation stack for compute machines:

Reviewer: s/ Creating the CloudFormation stack for compute machines should be a single step procedure module.

https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs

Reviewer: Also, add a short description for the module.

https://redhat-documentation.github.io/supplementary-style-guide/#shortdesc

> . To create the CloudFormation stack for compute machines:

Reviewer: ```suggestion
* To create the CloudFormation stack for compute machines, run the following command:
```

Reviewer: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command

Reviewer: https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs:~:text=as%20a%20comment.-,Procedure,-This%20section%20is

[source,terminal]
----
$ aws cloudformation create-stack --stack-name <name> <1>
     --template-body file://<template>.yaml \ <2>
     --parameters file://<parameters>.json <3>
----

> <1> Replace `<name>` with the name for the CloudFormation stack, such as `cluster-worker-1`. You need the name of this stack if you remove the cluster.

Reviewer: ```suggestion
<1> Specify the name for the CloudFormation stack, such as `cluster-worker-1`. You need the name of this stack if you remove the cluster.
```

Reviewer: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#callouts-in-code-blocks

> <2> The parameter `<template>` is the relative path and the name of the CloudFormation template YAML file that you saved.

Reviewer: ```suggestion
<2> Specify the relative path and the name of the CloudFormation template YAML file that you saved.
```

> <3> Defines the name of the`<parameters>`. The file name is the relative path and the name of the JSON file for the CloudFormation parameters.

Reviewer: ```suggestion
<3> Specify the relative path and the name of the JSON file for the CloudFormation parameters.
```

[id="installation-cloudformation-worker_{context}"]
= CloudFormation template for worker machines

> You can use the following CloudFormation template to deploy the worker machines that you need for your {product-title} cluster.

Reviewer 2: worker machines > compute machines?

See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#using-conscious-language

include::https://raw.githubusercontent.com/openshift/installer/release-4.18/upi/aws/cloudformation/06_cluster_worker_node.yaml[]
----
====
[IMPORTANT]
====

> When you use the CloudFormation template for the control plane machines, it provisions all three control plane machines with a single stack. However, when you use the CloudFormation template to deploy the worker machines, you must create the number of stacks based on the number that you defined in the `install-config.yaml` file. Each stack is provisioned one time only for each machine. To provision a new worker machine, you must change the stack name.

Reviewer 2: ```suggestion
When you use the CloudFormation template for the control plane machines, it provisions all three control plane machines with a single stack. However, when you use the CloudFormation template to deploy the compute machines, you must create the number of stacks based on the number that you defined in the `install-config.yaml` file. Each stack is provisioned one time only for each machine. To provision a new compute machine, you must change the stack name.
```

====

> [source,terminal]

Reviewer 2: This command could do with a lead-in sentence as the context of the command dangles with little context after a big configuration block.

----
$ aws cloudformation create-stack --stack-name <name> <1>
     --template-body file://<template>.yaml \ <2>
     --parameters file://<parameters>.json <3>
----
<1> Replace `<name>` with the name for the CloudFormation stack, such as `cluster-worker-1`. You need the name of this stack if you remove the cluster.

> <2> The parameter `<template>` is the relative path to and the name of the CloudFormation template YAML file that you saved.

Reviewer 2: ```suggestion
<2> The parameter `<template>` is the relative path and the name of the CloudFormation template YAML file that you saved.
```

include::https://raw.githubusercontent.com/openshift/installer/release-4.18/upi/aws/cloudformation/06_cluster_worker_node.yaml[]
----
====
[IMPORTANT]
====

> When you use the CloudFormation template for the master machines, it provisions all three master machines with a single stack. However, when you use the CloudFormation template to deploy the worker machines, you must create the number of stacks based on the number that you defined in the `install-config.yaml` file. Each stack is provisioned one time only for each machine. To provision a new worker machine, you must change the stack name.

Reviewer 3: ```suggestion
When you use the CloudFormation template for the control plane machines, it provisions all three master machines with a single stack. However, when you use the CloudFormation template to deploy the worker machines, you must create the number of stacks based on the number that you defined in the `install-config.yaml` file. Each stack is provisioned one time only for each machine. To provision a new worker machine, you must change the stack name.
```

https://redhat-documentation.github.io/supplementary-style-guide/#control-plane-node 
IMO control plane can be used here but you might want to confirm with QE or SME

> When you use the CloudFormation template for the master machines, it provisions all three master machines with a single stack. When you use the CloudFormation template to deploy the worker machines, however, you must create the number of stacks based on your requirements, that is, the number you defined in the `install-config.yaml` file. Each stack is provisioned one time only for each machine. To provision a new worker machine, you must change the stack name.

Reviewer 4: ```suggestion
When you use the CloudFormation template for the master machines, it provisions all three master machines with a single stack. However, when you use the CloudFormation template to deploy the worker machines, you must create the number of stacks based on the number that you defined in the `install-config.yaml` file. Each stack is provisioned one time only for each machine. To provision a new worker machine, you must change the stack name.
```
Trying to suggest something that makes the sentence a bit less long and wordy, but I don't know if it's really improved. Use if it's useful.

====

[source,terminal]
----
$ aws cloudformation create-stack --stack-name <name> <1>
     --template-body file://<template>.yaml \ <2>
     --parameters file://<parameters>.json <3>
----
<1> `<name>` is the name for the CloudFormation stack, such as `cluster-worker-1`. You need the name of this stack if you remove the cluster.
<2> `<template>` is the relative path to and name of the CloudFormation template YAML file that you saved.
<3> `<parameters>` is the relative path to and name of the CloudFormation

> parameters JSON file.

Reviewer 3: As per guideline[1], we avoid starting any sentence with lowercase. So consider updating all call out descriptions with something like
Option 1:  -- "Replace `<name>` with the name of...." 
Option 2: -- "The `<name> parameter....."
Option 3: -- "Defines the name of the....."

[1] https://www.ibm.com/docs/en/ibm-style?topic=grammar-capitalization#general-text

> Do not start a sentence with a lowercase term; rewrite the sentence.
